firmware/application/apps/ui_looking_glass_app.cpp

```diff
@@ -251,17 +250,24 @@ void GlassView::on_range_changed() {
         bin_length = 80;
         ignore_dc = 0;
     }
-    each_bin_size = looking_glass_bandwidth / SPEC_NB_BINS;
-    if (mode != LOOKING_GLASS_SINGLEPASS) {
-        looking_glass_step = (bin_length + (ignore_dc * 12)) * each_bin_size;
-    } else {
-        looking_glass_step = SPEC_NB_BINS * each_bin_size;
-    }
+
     adjust_range(&f_min, &f_max, SCREEN_W);
     looking_glass_range = f_max - f_min;
-    marker_pixel_step = looking_glass_range / SCREEN_W;  // Each pixel value in Hz
+    if (mode == LOOKING_GLASS_SINGLEPASS) {
+        looking_glass_bandwidth = looking_glass_range;
+        looking_glass_sampling_rate = looking_glass_bandwidth / 2;
+        each_bin_size = looking_glass_bandwidth / bin_length;
+        looking_glass_step = bin_length * each_bin_size;
+    } else {  // if ( mode == LOOKING_GLASS_SLOWSCAN || mode == LOOKING_GLASS_FASTSCAN )
```

Comment:
Does this comment still need to be here?

Reply:
I left it to indicate to two implicit condition